racket/collects/racket/match/parse.rkt

```diff
@@ -128,6 +257,30 @@
                   (rearm+parse (syntax/loc stx (list es ...))))]
     [(vector es ...)
      (Vector (map rearm+parse (syntax->list #'(es ...))))]
+
+    ;; Hash table patterns
+    [(hash* kvp ... #:rest rest-pat)
+     (eq? (syntax-e #'rest-pat) '_)
+     (do-hash stx #'(kvp ...) #f)]
+    [(hash* kvp ... #:rest rest-pat) (do-hash stx #'(kvp ...) #'rest-pat)]
+    [(hash* kvp ... #:closed) (do-hash stx #'(kvp ...) #t)]
+    [(hash* kvp ... #:open) (do-hash stx #'(kvp ...) #f)]
+    [(hash* kvp ...) (do-hash stx #'(kvp ...) #f)]
+
+    [(hash es ... #:rest rest-pat)
```

Comment:
It's better to call `do-hash` directly here rather than going through `parse` again.

Reply:
I agree for `#:closed` and `#:open`, but for `#:rest`, I think it makes sense to go through `parse` again because it handles `_` specially, and I'd rather not duplicate that logic here.